Allow running stylua as an optional step

ColinKennedy opened the pull request:
Closes: https://github.com/JohnnyMorganz/stylua-action/issues/57

Example:
```yml
      - uses: JohnnyMorganz/stylua-action@vFutureReleaseAfterThisPrIsMerged
        with:
          version: latest
          token: ${{ secrets.GITHUB_TOKEN }}
          args: false
```

JohnnyMorganz commented:
Thanks, this makes sense to me!

Do you mind adding a quick smoketest to the github CI check with `args: false` example? (And then a later step which uses the cached artifact)

Also, could you update the README?

ColinKennedy commented:
I added a smoketest that adds a file that contains a stylua issue then used `cmp` to verify that the file is unchanged. Will that work as a test? I've also updated the README.md

ColinKennedy commented:
I call stylua indirectly, via make. See https://github.com/ColinKennedy/nvim-best-practices-plugin-template/blob/main/Makefile. Repository structures tend to vary but using make as a bit of indirection means I can call `make check-stylua` or `make stylua` on any repo and it will always do the right thing, which I find that consistency useful. That's why

ColinKennedy commented:
> I assume it's available on the PATH so you can just reference stylua in a run step?

Yes, I call stylua indirectly, via make. See https://github.com/ColinKennedy/nvim-best-practices-plugin-template/blob/main/Makefile. Repository structures tend to vary but using make as a bit of indirection means I can call `make check-stylua` or `make stylua` on any repo and it will always do the right thing, which I find that consistency useful. That's why

ColinKennedy commented:
@JohnnyMorganz I'm not sure what caused https://github.com/JohnnyMorganz/stylua-action/actions/runs/12738986339/job/35542812438?pr=58 to fail but in case it was because of a difference in newline, I removed that code. Would you let me know if you need me to do anything else for this PR?

JohnnyMorganz commented:
Looks like it might be line endings issues. I can take a look at that on the weekend, but otherwise LGTM

ColinKennedy commented:
@JohnnyMorganz any chance we can move forward with this PR?
